app/assets/javascripts/models/beta/projectBoard.js

```diff
@@ -114,16 +142,32 @@ export const moveStory = (
   destinationArray,
   sourceIndex,
   destinationIndex,
+  isSameColumn
 ) => {
   const newSourceArray = [...sourceArray];
   const [removed] = newSourceArray.splice(sourceIndex, 1);
-  const newDestinationArray = [...destinationArray];
-  const filteredArray = newDestinationArray.filter((el, index) => index !== sourceIndex);
-  filteredArray.splice(destinationIndex, 0, removed);
 
-  return [...filteredArray];
+  let newDestinationArray = [...destinationArray];
+  if (isSameColumn) {
+    newDestinationArray = newDestinationArray.filter((el, index) => index !== sourceIndex);
+  }
+  newDestinationArray.splice(destinationIndex, 0, removed);
+  const sortedStories = sortStories(newDestinationArray, destinationIndex, isSameColumn);
+
+  return [...sortedStories];
 };
 
+export const sortStories = (destinationArray, destinationIndex, isSameColumn) => {
+  const newDestinationArray = [...destinationArray];
+  const formatedArray = newDestinationArray.map((item, index) => {
+      if (index >= destinationIndex) {
+        item.newPosition = item.newPosition + 1
+      };
+    return item;
+  });
+  return [...formatedArray];
+}
```

Comment:
this function could be improved like this:
```js
export const sortStories = (destinationArray, destinationIndex, isSameColumn) => {
  return destinationArray.map((item, index) => {
      if (index >= destinationIndex) {
         return { ...item, newPosition: item.newPosition + 1 }
      };
    return item;
  });
}
```

You dont need to create a new array and mutate it on the map

Reply:
Also, this function is receiving a `isSameColumn` argument that isnt being used right? That could be removed